docroot/modules/humsci/hs_paragraph_types/hs_paragraph_types.module

```diff
@@ -58,3 +58,56 @@ function hs_paragraph_types_field_widget_complete_paragraphs_form_alter(array &$
     }
   }
 }
+
+/**
+ * Implements hook_field_widget_complete_WIDGET_TYPE_form_alter().
+ */
+function hs_paragraph_types_field_widget_complete_paragraphs_browser_form_alter(array &$field_widget_form, FormStateInterface $form_state, array $context) {
+  // There are multiple issues with setting and removing the required state
+  // from paragraph fields when using the `conditional_fields` module, so
+  // use form alter to use the states api instead.
+  // @see https://www.drupal.org/project/conditional_fields/issues/3344587
+  // Check that hs_collection is the correct paragraph type
+  // and correct fields are present.
+  if (!empty($field_widget_form['widget']['#field_name']) && $field_widget_form['widget']['#field_name'] == 'field_hs_page_components') {
+    foreach (Element::children($field_widget_form['widget']) as $key) {
+      if (!is_int($key)) {
+        continue;
+      }
+      $element = $field_widget_form['widget'][$key];
+      if (
+        !empty($element['#paragraph_type']) && $element['#paragraph_type']== 'hs_collection' &&
+        !empty($element['subform']['field_title']) &&
+        !empty($element['subform']['field_title_settings'])
+      ) {
+        // Build title settings name for this component.
+        $title_settings_name = '';
+        foreach ($element['subform']['field_title_settings']['widget']['#field_parents'] as $delta) {
+          if (!empty($title_settings_name)) {
+            $delta = "[$delta]";
+          }
+          $title_settings_name .= $delta;
+        }
+        $title_settings_name .= '[field_title_settings]';
+
+        $header_conditions = [
```

Comment:
What is this doing that the conditional_fields can't do?

Reply:
We're using Forms API for the conditionals because there are some issues with  how `conditional_fields` handles changing the "Required" state:
- https://www.drupal.org/project/conditional_fields/issues/2859667
- https://www.drupal.org/project/conditional_fields/issues/3344587

Reply:
Ok thanks. Just wanted to make sure ya'll checked.